### PR TITLE
Fix option_list methods handling of nils

### LINKED ISSUE #59: option_list doesn't handle nil
If you call option_list on a nil or undefined object, it throws an error. 

```
           ================================================================================
           Error executing action `register` on resource 'octopus_deploy_tentacle[Tentacle]'
           ================================================================================

           NoMethodError
           -------------
           undefined method `map' for nil:NilClass

           Resource Declaration:


```
option_list is only currently used in a couple of places, and those are all required attributes. 

Make it so that option_list can handle optional attributes (nil)

### PR DESCRIPTION
**Pull request by spuder**
### Description

If `option_list` function in libraries/shared.rb is given a non existant attribute, it will cause chef run to fail. 
By allowing option_list to be given nil parameters, it allows optional attributes to be defined. (e.g tenant and tentanttag which will be arriving in another pull request). 

### Issues Resolved

#59 

### Contribution Check List
- [x] All tests pass.
- [x] New functionality includes testing.

### CODE REVIEW

**Comment by brentm5** [comment]
You just have one error in travis.

**Comment by spuder** [comment]
After more consideration, I don't think this is the best way to handle this.
Better would be to validate that you are passing the correct data type `option_list`. Functions that sometimes return strings and other times return nil objects just feels wrong to me. 

Alternative solution is only call option_list if you have a non nil object

    #{tenant.nil? ? '' : option_list('tenant', tenant) }

**Comment by brentm5** [comment]
I think its a valid use case to have either a nil or empty list of options however not a nil name.

**Comment by spuder** [comment]
Reopened & fixed rubocop

Map won't work against nil


```
puts option_list('foo')
ArgumentError: wrong number of arguments (1 for 2)

puts option_list('foo', nil)
NoMethodError: undefined method `map' for nil:NilClass
```

File: spec/unit/lib_shared_spec.rb [hunk 1]
```diff
@@ -17,6 +17,12 @@
       options = %w(option1 option2)
       expect(shared.option_list(name, options)).to eq '--name "option1" --name "option2"'
     end
+
+    it 'should return nil if nil parameter provided' do
+      name = nil
+      options = '42'
```
Comment (brentm5) [inline spec/unit/lib_shared_spec.rb]: i would check that the options can also nil

**Comment by brentm5** [comment]
The reason why the following fails is because we have not provided a default. 

```ruby
puts option_list('foo')
ArgumentError: wrong number of arguments (1 for 2)
```

we could do the following:

``` ruby
def option_list(name, options = [])
 <code>
end
```

**Comment by spuder** [comment]
That probably wasn't the best example. If a function requires 2 parameters, it should get both parameters. I can't think of any reason why someone would accidentally give it 1. 

The real error is when the second parameter is nil

```
puts option_list('foo', nil)
NoMethodError: undefined method `map' for nil:NilClass
```

File: spec/unit/lib_shared_spec.rb (hunk 1 above)
Comment (brentm5) [inline spec/unit/lib_shared_spec.rb]: I think the reverse of this would make more sense, so test that name is something and options is nil that way it verifies the test case we are trying to verify

**Comment by spuder** [comment]
Added another test and resquashed. Should be ready to :shipit:
